tests/test_xlayer_cs.py

```diff
@@ -31,19 +33,48 @@ def test_physical_path_has_ordered_list_of_involved_links(model: MelodyModel):
     assert actual == expected
 
 
-def test_physical_path_has_exchanges(model: MelodyModel):
+def test_PhysicalPath_has_exchanges(model: MelodyModel):
     exchange = model.pa.all_component_exchanges.by_name("C 6")
     path = model.pa.all_physical_paths.by_name("card1 - card2 connection")
     assert path.exchanges == [exchange]
 
 
-def test_physical_link_has_physical_paths(model: MelodyModel):
+def test_PhysicalLink_has_physical_paths(model: MelodyModel):
     link = model.pa.all_physical_links.by_name("Eth Cable 2")
     path = model.pa.all_physical_paths.by_name("card1 - card2 connection")
     assert link.physical_paths == [path]
 
 
-def test_physical_link_has_exchanges(model: MelodyModel):
+def test_PhysicalLink_has_exchanges(model: MelodyModel):
     link = model.pa.all_physical_links.by_name("Eth Cable 2")
     exchange = model.pa.all_component_exchanges.by_name("C 3")
     assert link.exchanges == [exchange]
+
+
+@pytest.mark.parametrize("attr_name", ["ends", "linkEnds"])
+def test_PhysicalLink_setting_ends(model: MelodyModel, attr_name: str):
+    link = model.pa.all_physical_links.by_name("Eth Cable 2")
+    source_pp = model.by_uuid("76d9c301-c0ad-4615-9f02-b804b018decf")
+    target_pp = model.by_uuid("53c9fe29-18e2-4642-906e-b7507bf0ff39")
+    expected = [source_pp, target_pp]
+    assert getattr(link, attr_name, None) != expected
+
+    setattr(link, attr_name, [source_pp, target_pp])
+    actual = getattr(link, attr_name)
+
+    assert actual == expected
+    assert [link.source, link.target] == expected
+
+
```

Comment:
```suggestion
    old_ends = getattr(link, attr_name, None)

    setattr(link, attr_name, [source_pp, target_pp])
    actual = getattr(link, attr_name)

    assert old_ends != expected
    assert actual == expected
    assert [link.source, link.target] == expected

```
I like this more b/c of the prepare, act, assert pattern/convention.

Reply:
I'd rather not, for the same reason I outlined in [another comment](https://github.com/DSD-DBS/py-capellambse/pull/182#discussion_r1011508921).